convert2rhel/checks.py

```diff
@@ -30,6 +32,7 @@
 
 logger = logging.getLogger(__name__)
 
+PACKAGE_NAMES_RE = r"^[0-9]*:?([a-z][a-z0-9-_]*?)(-[0-9])*\."
```

Comment:
One problem with regexes is that they are generally slower than other ways of checking.  Assigning regexes outside of any potential code loops like you're doing here is one way to mitigate that.  However, you need to not just assign the string here but also compile the regex (the speed savings is because you only have to compile the regex once, when the module is first imported).   So do something like this:
```suggestion
PACKAGE_NAMES_RE = re.compile(r"^[0-9]*:?([a-z][a-z0-9-_]*?)(-[0-9])*\.")
```
```

Reply:
Hmm.. also... does this regex work for the data we're being given?  I was going to make a suggestion about the other problem with regexes being that they're hard to debug so we should comment new regexes heavily.  But when I tried running some yum check-output data through it so I could understand what this regex is doing, I came out with zero matches instead of the one match I was expecting.

``` python 
>>> import re
>>> import sh  # amoffat's sh library
>>> a = re.compile(r"^[0-9]*:?([a-z][a-z0-9-_]*?)(-[0-9])*\.")
>>> yum_stdout = sh.yum(['check-update', '-q'], _ok_code=[0, 100])
>>> yum_stdout

python3-pyOpenSSL.noarch              21.0.0-1.fc33              updates-testing
Security: kernel-core-5.14.15-100.fc33.x86_64 is an installed security update
Security: kernel-core-5.14.13-100.fc33.x86_64 is the currently running version

>>> re.findall(a, str(yum_stdout))
[]
```

Reply:
@abadger I've tested with the 2 ways, and now (yesterday I got the same output as you, an empty []) they are working as expected. I've changed to your suggestions (compiling the regex first) 

```python
('packages', [(u'binutils', u''), (u'convert2rhel', u''), (u'glibc', u''), (u'glibc-common', u''), (u'grub2', u''), (u'grub2-common', u''), (u'grub2-pc', u''), (u'grub2-pc-modules', u''), (u'grub2-tools', u''), (u'grub2-tools-extra', u''), (u'grub2-tools-minimal', u''), (u'kernel', u''), (u'kernel-tools', u''), (u'kernel-tools-libs', u''), (u'nfs-utils', u''), (u'nspr', u''), (u'nss', u''), (u'nss-softokn', u''), (u'nss-softokn-freebl', u''), (u'nss-sysinit', u''), (u'nss-tools', u''), (u'nss-util', u''), (u'openssl', u''), (u'openssl-libs', u''), (u'python-perf', u''), (u'sudo', u''), (u'tzdata', u'')])
```

And the log message

```bash
CRITICAL - The system has 27 packages to be updated. To proceed with the conversion, update the packages on your system by executing the following step:
```

Reply:
Interesting.  Can you post your output from `yum check-update` or explain how the regex works?  I'm seeing that the unittests also appear to be failing in the same way as my local attempts are... zero packages are found in the output.

Reply:
As @bocekm stated in another thread here, we decided to go with the manual approach for that validation as it is handling better in some other cases for `yum check-update -q` command. 

And as far as the regex, I don't really know why the outputs were different. I'm not an expert on regex, but I think it was something along with the regex compilation method. Don't really know what caused the empty `[]` on the output.